packages/smithy-client/src/date-utils.ts

```diff
@@ -31,5 +34,267 @@ export function dateToUtcString(date: Date): string {
+  if (value === null || value === undefined) {
+    return undefined;
+  }
+
+  let valueAsDouble: number;
+  if (typeof value === "number") {
+    valueAsDouble = value;
+  } else if (typeof value === "string") {
+    // TODO: remove when https://github.com/aws/aws-sdk-js-v3/pull/2729 is merged, maybe
+    if (!VALID_NUMBER.exec(value)) {
+      throw new TypeError("Epoch timestamps must be valid floating point numbers.");
+    }
+    valueAsDouble = strictParseDouble(value)!;
+  } else {
+    throw new TypeError("Epoch timestamps must be expressed as floating point numbers or their string representation");
+  }
+
+  if (isNaN(valueAsDouble) || valueAsDouble === Infinity || valueAsDouble === -Infinity) {
+    throw new TypeError("Epoch timestamps must be valid, non-Infinite, non-NaN numerics");
+  }
+  return new Date(Math.round(valueAsDouble * 1000));
+};
+
+/**
+ * Build a date from a numeric year, month, date, and an match with named groups
+ * "H", "m", s", and "frac", representing hours, minutes, seconds, and optional fractional seconds.
+ * @param year numeric year
+ * @param month numeric month, 1-indexed
+ * @param day numeric year
+ * @param match match with groups "H", "m", s", and "frac"
+ */
+const buildDate = (year: number, month: number, day: number, match: RegExpMatchArray): Date => {
+  const adjustedMonth = month - 1; // JavaScript, and our internal data structures, expect 0-indexed months
+  validateDayOfMonth(year, adjustedMonth, day);
+  // Adjust month down by 1
+  return new Date(
+    Date.UTC(
+      year,
+      adjustedMonth,
+      day,
+      parseDateValue(match.groups!["H"]!, "hour", 0, 23),
+      parseDateValue(match.groups!["m"]!, "minute", 0, 59),
+      // seconds can go up to 60 for leap seconds
+      parseDateValue(match.groups!["s"]!, "seconds", 0, 60),
+      parseMilliseconds(match.groups!["frac"])
+    )
+  );
+};
+
+/**
+ * RFC 850 dates use a 2-digit year; start with the assumption that if it doesn't
+ * match the current year, then it's a date in the future, then let adjustRfc850Year adjust
+ * the final date back to the past if it's too far in the future.
+ *
+ * Example: in 2021, start with the assumption that '11' is '2111', and that '22' is '2022'.
+ * adjustRfc850Year will adjust '11' to 2011, (as 2111 is more than 50 years in the future),
+ * but keep '22' as 2022. in 2099, '11' will represent '2111', but '98' should be '2098'.
+ * There's no description of an RFC 850 date being considered too far in the past in RFC-7231,
+ * so it's entirely possible that 2011 is a valid interpretation of '11' in 2099.
+ * @param value the 2 digit year to parse
+ * @return number a year that is equal to or greater than the current UTC year
+ */
+const parseTwoDigitYear = (value: string): number => {
+  const thisYear = new Date().getUTCFullYear();
+  const valueInThisCentury = Math.floor(thisYear / 100) * 100 + strictParseShort(value)!;
```

Comment:
You definitely can't use `strictParseShort` here - `02` is valid for 2002 for example but would be rejected for the leading 0.